spec/services/applicants/toggle_archive_applicant_spec.rb

```diff
@@ -0,0 +1,106 @@
+describe Applicants::ToggleArchiveApplicant, type: :service do
+  subject do
+    described_class.call(
+      rdv_solidarites_session: rdv_solidarites_session,
+      applicant: applicant, archiving_reason: archiving_reason,
+      archived_at: archived_at
+    )
+  end
+
+  let!(:applicant) { create(:applicant) }
+  let(:rdv_solidarites_session) { instance_double(RdvSolidaritesSession) }
+  let!(:archiving_reason) { "some reason" }
+  let!(:archived_at) { Time.zone.now }
+  let!(:invitation1) { create(:invitation, applicant: applicant) }
+  let!(:invitation2) { create(:invitation, applicant: applicant) }
+
+  describe "#call" do
+    before do
+      allow(applicant).to receive(:assign_attributes)
+        .with(archived_at: archived_at, archiving_reason: archiving_reason)
+      allow(applicant).to receive(:save)
+        .and_return(true)
+      allow(InvalidateInvitationJob).to receive(:perform_async)
+    end
+
+    it "is a success" do
+      expect(subject.success?).to eq(true)
+    end
+
+    it "tries to update the applicant" do
+      expect(applicant).to receive(:assign_attributes)
+        .with(archived_at: archived_at, archiving_reason: archiving_reason)
+      expect(applicant).to receive(:save)
+      subject
+    end
+
+    # it "change the archived_at value" do
+    #   subject
+    #   expect(applicant.reload.archived_at).to eq(archived_at)
+    # end
```

Comment:
Si tu mock les transactions active record tu peux pas tester qu'elles s'effectuent bien. 
Si tu veux tu peux garder ces tests et enlever toute la logique de mocking au dessus (`allow/expect to receive assign_attributes/save`).

Reply:
Merci !